packages/@lwc/engine-server/src/types.ts

```diff
@@ -38,15 +39,15 @@ export interface HostAttribute {
 }
 
 export interface HostShadowRoot {
-    type: HostNodeType.ShadowRoot;
+    [HostTypeAttr]: HostNodeType.ShadowRoot;
     children: HostChildNode[];
     mode: 'open' | 'closed';
     delegatesFocus: boolean;
 }
 
 export interface HostElement {
-    type: HostNodeType.Element;
-    name: string;
+    [HostTypeAttr]: HostNodeType.Element;
```

Comment:
While this PR fixes name clashing with `type` and `name`. We would still run into issues if a component ever defines public properties named `namespace`, `parent`, `shadowRoot`, `children`, `attributes` or `eventListeners`.

We should go a step further and harden our implementation for public props to not conflict with `HostElement` internal properties.

Reply:
I see that we do pass these elements to `engine-core` to generate and manage the VM. If we revamp all internal properties to use Symbols, would it require changes in both `engine-server` and `engine-core`? Or are references to the internal properties sufficiently isolated to the engine-server project?